hack/unify-job-properties.py

```diff
@@ -0,0 +1,164 @@
+
+    context = job.get('context')
+    if not context:
+        raise ValueError('job has no context')
+
+    return (branch, context)
+
+
+def get_siblings(job, jobs, suspect_branches):
+    try:
+        branch, context = job_branch_context(job=job)
+    except ValueError:
+        return None
+
+    siblings = {}
+    for sibling in jobs:
+        try:
+            sibling_branch, sibling_context = job_branch_context(job=sibling)
+        except ValueError:
+            continue
+        if sibling_branch in suspect_branches:
+            continue
+        if sibling_context == context and sibling_branch != branch:
+            siblings[sibling_branch] = sibling
+    return siblings
+
+
+def unify_job(job, siblings, mutable_properties=None):
+    if mutable_properties is None:
+        # https://docs.ci.openshift.org/docs/how-tos/contributing-openshift-release/#tolerated-changes-to-generated-jobs
+        mutable_properties = {
+            'always_run',
+            'max_concurrency',
+            'optional',
+            'reporter_config',
+            'run_if_changed',
+            'skip_if_only_changed',
+            'skip_report',
+        }
+    branch, context = job_branch_context(job=job)
+
+    updated = False
+    for prop in mutable_properties:
+        sibling_values = {json.dumps(sibling.get(prop), sort_keys=True) for sibling in siblings.values()}
+        if len(sibling_values) > 1:
+            print('cannot unify {} for {} {}.  Sibling values include:'.format(prop, branch, context))
+            for sibling_branch, sibling in sorted(siblings.items()):
+                print('  {}: {}'.format(sibling_branch, json.dumps(sibling.get(prop), sort_keys=True)))
+        elif len(sibling_values) == 1:
+            sibling_value = list(sibling_values)[0]
+            value = json.dumps(job.get(prop), sort_keys=True)
+            if value != sibling_value:
+                print('{} {}: change {} from {} to {} to match siblings'.format(
+                    branch, context, prop, value, sibling_value))
+                job[prop] = json.loads(sibling_value)
+                updated = True
+    if updated:
+        update_job(job=job)
+
+
+def update_job(job):
+    with open(job['_path']) as f:
+        data = yaml.load(f, yaml.RoundTripLoader)
+    if job['_type'] == 'periodics':
+        jobs = data[job['_type']]
+    else:
+        jobs = data[job['_type']][job['_repo']]
+    for i, j in enumerate(jobs):
+        if j['name'] == job['name']:
+            jobs[i] = {k: v for k,v in job.items() if not k.startswith('_')}
+            break
+    with open(job['_path'], 'w') as f:
+        yaml.dump(data, f, default_flow_style=False, Dumper=yaml.RoundTripDumper)
+
+
+if __name__ == '__main__':
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description='Unify 4.8 and 4.9 job configuration by adjusting mutable properties (optional, etc.) when they diverge from the master, 4.6, and 4.7 configuration.',
```

Comment:
how does it decide which branch to compare 4.8+4.9 to?  (master vs 4.6 vs 4.7?)

Reply:
it only automatically updates if all of 4.6, 4.7, and master agree on the mutable value for that job

Reply:
otherwise it just logs the inconsistency in case you want to manually adjust

Reply:
got it